src/routes/workspaces.py

```diff
@@ -0,0 +1,183 @@
+"""
+Workspace Management Routes for the MinIO Manager API.
+
+This module provides user workspace management including personal directories,
+group spaces, and workspace organization. Focuses on the user experience of
+organizing and managing their data storage.
+"""
+
+import logging
+import os
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, Path
+from pydantic import BaseModel, ConfigDict, Field
+
+from ..minio.core.minio_client import MinIOClient
+from ..minio.managers.group_manager import GroupManager
+from ..minio.managers.policy_manager import PolicyManager
+from ..minio.managers.user_manager import UserManager
+from ..minio.models.minio_config import MinIOConfig
+from ..minio.models.user import UserModel
+from ..service.arg_checkers import not_falsy
+from ..service.dependencies import auth
+from ..service.exceptions import MinIOManagerError
+from ..service.kb_auth import KBaseUser
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(prefix="/workspaces", tags=["workspaces"])
+
+
+# ===== RESPONSE MODELS =====
+
+
+class GroupWorkspaceResponse(BaseModel):
+    """Response model for group workspace information."""
+
+    model_config = ConfigDict(str_strip_whitespace=True, frozen=True)
+
+    group_name: Annotated[str, Field(description="Group name", min_length=1)]
+    members: Annotated[list[str], Field(description="Group members")]
+    member_count: Annotated[int, Field(description="Number of members", ge=0)]
+    accessible_paths: Annotated[
+        list[str], Field(description="Group's accessible paths")
+    ]
+
+
+class UserGroupsResponse(BaseModel):
+    """Response model for user's group memberships."""
+
+    model_config = ConfigDict(str_strip_whitespace=True, frozen=True)
+
+    username: Annotated[str, Field(description="Username", min_length=1)]
+    groups: Annotated[list[str], Field(description="Groups user belongs to")]
+    group_count: Annotated[int, Field(description="Number of groups", ge=0)]
+
+
+# ===== DEPENDENCY INJECTION =====
+
+
+async def get_workspace_managers():
+    """Get initialized managers for workspace operations."""
+    config = MinIOConfig(
+        endpoint=not_falsy(os.getenv("MINIO_ENDPOINT"), "MINIO_ENDPOINT"),
+        access_key=not_falsy(os.getenv("MINIO_ROOT_USER"), "MINIO_ROOT_USER"),
+        secret_key=not_falsy(os.getenv("MINIO_ROOT_PASSWORD"), "MINIO_ROOT_PASSWORD"),
+    )
+
+    client = MinIOClient(config)
+    await client._initialize_session()
```

Comment:
Calling a private method is a code smell

Reply:
`await MinioClient.initialize_session(config)` seems like the way to go

Reply:
👍